### ISSUE TITLE
Allow separate edit and display titles for field

### ISSUE THREAD

**Issue by bradbulger** [backdrop/backdrop-issues#4310]
On sites where users manage content types themselves, I frequently see fields with titles that include explanatory text - "Proposal Summary (Be specific - 250 characters maximum)". As they see it, their users don't see or pay attention to help text, and only putting the instructions or descriptions into the boldfaced field title will do.

Which is fine, but then that becomes the "name" of the field in other contexts, particularly in display. And we end up writing views to display the nodes because the users want to customize the titles used in display.

It might be useful to be able to customize the field title as a configuration option for the entity display.

There is a Drupal 7 contributed module, [Field formatter label](https://www.drupal.org/project/field_formatter_label), that looks like it does this. It's dependent on another contributed module, [Field formatter settings](https://www.drupal.org/project/field_formatter_settings), that has since been incorporated into core for Drupal 8.

So maybe it would be best to address this issue in a similar way. Depends on how common the need is, I suppose.

**Comment by klonos** [comment]
The usage count for [Field formatter settings](https://www.drupal.org/project/field_formatter_settings) is currently close to 50k, which is pretty decent; but [Field formatter label](https://www.drupal.org/project/field_formatter_label) has a count of less than 350 sites as it is. ...so I would say this is not such a common request @bradbulger 😅

Perhaps in contrib?

Another way to solve this would be an admin subtheme, where the field labels are overridden to include the actual label + the help text.

**Comment by klonos** [comment]
...I just tested this really quick: the help fields when configuring fields support certain HTML tags, and allow for classes to be added. So I have created a field, with this as help text: `<strong class="red">Be specific - 250 characters maximum<strong>` ...with a single line of CSS, I was able to then have it render like so:

![Screen Shot 2020-02-13 at 7 43 52 am](https://user-images.githubusercontent.com/2423362/74375666-dcece500-4e34-11ea-9cc3-f2bd9f2627c4.png)

**Comment by bradbulger** [comment]
That's a solution for some people, but others really only will accept being able to write full English questions as the field label for the create/edit form.

Maybe reverse the process that Field formatter settings took, do it as contrib first, and then maybe it could be moved into core later? A port of Field formatter label sounds like the right approach for that issue specifically.

**Comment by klonos** [comment]
Yes, I think that contrib is the right way to go. Not sure if it will get enough traction/usage to justify it being merged into core though.

**Comment by olafgrabienski** [comment]
> but Field formatter label has a count of less than 350 sites

That's correct but there are way more popular modules doing the same thing (in another way or as part of a set of more features):

- https://www.drupal.org/project/field_display_label (7,744 sites even if unsupported module status)
- https://www.drupal.org/project/ds (Field template UI of Display Suite, 156,731 sites)

The latter reminds me on a thread I opened long time ago (https://github.com/backdrop/backdrop-issues/issues/2084) where @quicksketch [stated](https://github.com/backdrop/backdrop-issues/issues/2084#issuecomment-244277095):

> Overriding the label text and specify customized HTML...  Right now the ability to customize field display is pretty limited (just formatter settings). You're right that we want to allow for more control over markup through the UI and Field module lags behind Layout and Views modules in this regard.

Still true, in my opinion. There are many quite common use cases where people could need that. I'll mention a very common one here (from https://www.drupal.org/project/field_display_label):

> For example, if you want the field to read "What is your name?" when editing content, but to read "Name" when viewing the content.

So, in my opinion this should be a direct candidate for core. If instead there will be a good contrib alternative before, I'm confident it will get a lot of traction for inclusion in core.

**Comment by olafgrabienski** [comment]
@bradbulger Have you seen https://backdropcms.org/project/field_wrappers? It allows to choose HTML wrappers for fields, so a very similar use case (adjusting field markup via the user interface). I'm wondering if it would make sense to integrate the label manipulation feature in that module, so that people don't have to install two modules when configuring field display via the UI. What do you think about it?

**Comment by BWPanda** [comment]
Field Formatter Settings is in core: #1376

**Comment by olafgrabienski** [comment]
> Field Formatter Settings is in core

What does it mean, in this case? (I guess it doesn't change anything directly but helps in developing a solution.)

**Comment by BWPanda** [comment]
Just means you don't need to have a contrib dependency :slightly_smiling_face:

**Comment by bradbulger** [comment]
@olafgrabienski that Field Wrappers project does seem similar but this is a pretty significant addition - maybe build on it as a dependency?

the advantage of Field formatter label, to my mind, is that it lets you create different labels for different display modes. I can see that adding it to the field config is simpler. depends on what people prefer.

**Comment by olafgrabienski** [comment]
> the advantage of Field formatter label, to my mind, is that it lets you create different labels for different display modes

@bradbulger FYI, [Field Wrappers](https://backdropcms.org/project/field_wrappers) is also configured at the *display* mode level which might be not clear when you read the project page. It allows to choose the HTML markup for the entire field, single field items and even the field label. Don't know if that makes it more interesting to build upon. Anyway, here's a screenshot of the user interface:

![screeen-backdrop-field-wrappers](https://user-images.githubusercontent.com/14188307/74518525-48f24a80-4f14-11ea-9e89-c97d9f38ebbd.png)
